src/Components/TrainingPage.js

```diff
@@ -9,74 +9,82 @@ export function getGoals(trainingDrills) {
 }
 
 export function getTrainingDuration(trainingDrills) {
-  const durationList = trainingDrills.map(({ durationInMinutes }) => durationInMinutes);
-  return durationList.reduce((a, b) => a + b, 0);
+  return trainingDrills.reduce((total, drill) => total + drill.durationInMinutes, 0);
 }
 
 export function getTrainingMinimalPlayersNumber(trainingDrills) {
   const minimalPlayersNumberList = trainingDrills.map(({ minimalPlayersNumber }) => minimalPlayersNumber);
   return Math.max(...minimalPlayersNumberList);
 }
 
-const mapStateToProps = state => {
-  return {
-    drills: state.drills,
-  };
-};
+export const TrainingPage = props => {
+  const { navigation, route, drills } = props;
+  const training = route.params.training;
+  const trainingDrills = training.drills.map(drillId => drills.find(drill => drill.id === drillId));
 
-export class TrainingPage extends Component {
-  render() {
-    const { navigation } = this.props;
-    const training = this.props.route.params.training;
-    const drills = training.drills.map(drillId => this.props.drills.find(drill => drill.id === drillId));
-
-    return (
-      <View style={styles.trainingPage}>
+  return (
+    <ScrollView style={styles.trainingPage} contentContainerStyle={styles.trainingPageContent}>
```

Comment:
Don't you get a warning using a FlatList inside this scrollview ? ?

Reply:
Ah, yes I got the warning. It seems to be related to performance issue on large lists of elements and I don't think we'll have more than 10 drills there.

But I'll look for a solution tomorrow

Reply:
I had a look at it this morning and it seems that:
- It is a performance issue because all the elements are rendered even if they are not visible on the screen. This is not so important here since training drills are < 10
- VDD is going to think about the training page and probably going to suggest something pretty different, so w'll have to refactor it later on anyways